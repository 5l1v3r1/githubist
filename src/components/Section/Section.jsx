// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import styles from './Section.scss';

type Props = {
  children: Node,
  secondary?: boolean,
  className?: string,
};

const Section = ({ children, secondary, className }: Props) => {
  const classNames = cx(styles.section, { [styles.secondary]: secondary }, className);

  return <section className={classNames}>{children}</section>;
};

Section.defaultProps = {
  secondary: false,
  className: '',
};

export default Section;
