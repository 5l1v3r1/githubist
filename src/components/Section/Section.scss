@import '../../theme/theme';

.section {
  padding: $section-gap 0;
}

.secondary {
  background-color: $hint-of-red;
}
