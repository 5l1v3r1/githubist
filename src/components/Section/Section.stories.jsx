// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';

import Section from './Section';

storiesOf('Section', module)
  .add('default', () => (
    <Section>
      <div>Section text</div>
    </Section>
  ))
  .add('secondary', () => (
    <Section secondary>
      <div>Section text</div>
    </Section>
  ));
