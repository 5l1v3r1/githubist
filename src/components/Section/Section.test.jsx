// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Section from './Section';

it('renders correctly', () => {
  const wrapper = shallow(
    <Section>
      <p>Hello world</p>
    </Section>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly secondary', () => {
  const wrapper = shallow(
    <Section secondary>
      <p>Hello world</p>
    </Section>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('passes custom classnames to image', () => {
  const wrapper = shallow(
    <Section className="testClassName">
      <p>Hello world</p>
    </Section>,
  );

  expect(wrapper.find('section').hasClass('testClassName')).toBe(true);
});
