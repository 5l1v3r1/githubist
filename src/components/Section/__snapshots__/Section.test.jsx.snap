// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className=""
>
  <p>
    Hello world
  </p>
</section>
`;

exports[`renders correctly secondary 1`] = `
<section
  className="undefined"
>
  <p>
    Hello world
  </p>
</section>
`;
